Automattic/jetpack
# WordPress 6.7 compatibility: Add __nextHasNoMarginBottom to controls to prevent deprecation notices

Fixes https://github.com/Automattic/jetpack/issues/39823

## Proposed changes:

* This PR adds `__nextHasNoMarginBottom={ true }` to multiple BaseControl-based components, to prevent deprecation notices ([more context about these here, and the changes generally](https://make.wordpress.org/core/2024/10/18/editor-components-updates-in-wordpress-6-7/#bottom-margin-styles-are-deprecated-39358)). As much as possible, the addition of this prop has been tested for each change. The intention here is to cleanup as many of the notices as possible prior to the next release - not all are covered:
  * In several cases I was unable to test features so to be safe have left those out (inc. VideoPress Video chapters (`DetailsControl`), `LikeEdit` component for Like block on Simple (as it only affects Simple), among a few others).
  * Some I was unable to test but the change was most likely needed, so added them. These were generally `ToggleControl` related, where all other tested examples had needed the addition. Examples include `visibilityAdvancedControls` related to the Widget Visibility module.
  * No changes were made to CRM Automations files, as that feature was never implemented.
  * I did not make any changes in the `jetpack-mu-wpcom` package, partially due to testing issues (eg. Newspack related blocks), and also due to releases on WoA being more frequent should anything obvious be spotted otherwise.
  * Deprecated versions of files
  * Also any component that has imported `ToggleControl` from `@automattic/jetpack-components` does not need changing, as the Jetpack component has the prop added.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/39823

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

**Self-hosted site test instructions**

To replicate the issue:

* On a test site, connected to WordPress.com,  install the WordPress Beta Tester plugin
* Go to Tools > Beta Testing, and set the plugin to use Beta/RC Only
* Go to Dashboard > Updates, and update to the most recent version of WP.
* Ensure script debug is enabled on your site - `define( 'SCRIPT_DEBUG', true );` in `wp-config.php`.
* Open up the post editor for a post, and in the browser console there should be several deprecation notices showing. 
* There are multiple places to test and check - browsing the code for changes files can help but as a quick guide the post editor, testing blocks (particularly the block settings sidebar, checking each panel, also the pre-publish panel and Jetpack setting panel), the Jetpack settings pages. Some blocks to test include Form blocks, Opentable, Tock, VideoPress, AI chat, Podcast player.
* Note you will see some 6.6 related notices too, they will be dealt with separately: https://github.com/Automattic/vulcan/issues/562

To test the changes:

* Apply this PR, making sure to rebuild any plugins / packages you are testing changes related to.
* Confirm the notices are gone, and for any specific features tested confirm there are no style adjustments (bottom margin related) that appear or look out of place.
